inference/app/src/exact/lc_names.py

```diff
@@ -0,0 +1,47 @@
+import os
+import requests
+
+
+def get_lc_names_api_response(lc_names_id):
+    url = f"http://id.loc.gov/authorities/names/{lc_names_id}.json"
+    api_response = requests.get(url)
+    if api_response.status_code == 200:
+        return api_response.json()
+    elif api_response.status_code == 404:
+        raise ValueError(f"{lc_names_id} is not a valid lc_names_id")
+    else:
+        raise ValueError(
+            f"something unexpected happened when calling url: {url}"
+        )
+
+
+def get_lc_names_preferred_label(lc_names_id, api_response=None):
+    api_response = api_response or get_lc_names_api_response(lc_names_id)
+    preferred_label_id = "http://www.w3.org/2004/02/skos/core#prefLabel"
+    preferred_label = None
+    for element in api_response:
+        if preferred_label_id in element:
+            preferred_label = element[preferred_label_id][0]["@value"]
+    return preferred_label
+
+
+def get_lc_names_variants(lc_names_id, api_response=None):
+    api_response = api_response or get_lc_names_api_response(lc_names_id)
+    variant_id = "http://www.loc.gov/mads/rdf/v1#variantLabel"
```

Comment:
This `#` isn't going to be read by the server, what is it intended to be doing?

Reply:
the string is a key in the LoC API's (fairly ugly) response - see http://id.loc.gov/authorities/names/nb2015021801.json for an example

Reply:
`#variantLabel` <= this has though, is it doing anything?

Reply:
we never send a request to `http://www.loc.gov/mads/rdf/v1#variantLabel`, it's literally just used as a key to get to the right bit of the json response

Reply:
Crumbles, not sure why I was getting so confused.

Reply:
no worries! i have no idea why the api is organised that way - seems bound to create exactly this sort of confusion